qiime2/sdk/util.py

```diff
@@ -92,3 +92,30 @@ def parse_format(format_str):
     except KeyError:
         raise TypeError("No format: %s" % format_str)
     return format_record.format
+
+
+def artifact_actions(string):
```

Comment:
Maybe `actions_by_input_type` would be a better name. It's a little over-specific, but I could see this functionality evolving into a full API with mutliple inputs/output query/solving in the future, so this should be pretty future-proof.